pulsar-transaction/buffer/src/main/java/org/apache/pulsar/transaction/buffer/impl/PersistentTransactionBuffer.java

```diff
@@ -0,0 +1,308 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.transaction.buffer.impl;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.io.Serializable;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.mledger.AsyncCallbacks;
+import org.apache.bookkeeper.mledger.ManagedCursor;
+import org.apache.bookkeeper.mledger.ManagedLedger;
+import org.apache.bookkeeper.mledger.ManagedLedgerException;
+import org.apache.bookkeeper.mledger.Position;
+import org.apache.bookkeeper.mledger.impl.PositionImpl;
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.pulsar.broker.service.BrokerService;
+import org.apache.pulsar.broker.service.BrokerServiceException;
+import org.apache.pulsar.broker.service.persistent.PersistentTopic;
+import org.apache.pulsar.common.util.FutureUtil;
+import org.apache.pulsar.transaction.buffer.TransactionBuffer;
+import org.apache.pulsar.transaction.buffer.TransactionBufferReader;
+import org.apache.pulsar.transaction.buffer.TransactionCursor;
+import org.apache.pulsar.transaction.buffer.TransactionMeta;
+import org.apache.pulsar.transaction.buffer.exceptions.TransactionNotSealedException;
+import org.apache.pulsar.transaction.impl.common.TxnID;
+import org.apache.pulsar.transaction.impl.common.TxnStatus;
+
+/**
+ * A persistent transaction buffer implementation.
+ */
+@Slf4j
+public class PersistentTransactionBuffer extends PersistentTopic implements TransactionBuffer {
+
+    private TransactionCursor txnCursor;
+    private ManagedCursor retentionCursor;
+
+
+    abstract class TxnCtx implements PublishContext{
+        private long sequenceId;
+
+        TxnCtx(long sequenceId) {
+            this.sequenceId = sequenceId;
+        }
+
+        @Override
+        public String getProducerName() {
+            return "txn-producer";
+        }
+
+        @Override
+        public long getSequenceId() {
+            return this.sequenceId;
+        }
+    }
+
+    @Builder
+    private static final class Marker implements Serializable {
+        TxnID txnID;
+        TxnStatus status;
+
+        public byte[] serialize() {
+            return SerializationUtils.serialize(this);
```

Comment:
`SerializationUtils` is using java object serialization framework. It is not recommended. Can you change to use Schema.JSON or Schema.AVRO for serializing it?